recipes/easybuild.py

```diff
@@ -0,0 +1,51 @@
+"""EasyBuild container (https://github.com/easybuilders/easybuild)
+   Set the user argument 'easyconfig' to specify the EasyConfig to
+   build.  Otherwise, it will just build a base EasyBuild container
+   image.
+
+   Sample workflow:
+$ hpccm.py --recipe recipes/easybuild.py --userarg easyconfig=GROMACS-2016.3-foss-2016b-GPU-enabled.eb > Dockerfile.gromacs.eb
+$ docker build -t gromacs.eb -f Dockerfile.gromacs.eb .
+$ nvidia-docker run --rm -ti gromacs.eb bash -l
+container:/tmp> module load GROMACS
+...
+"""
+# pylint: disable=invalid-name, undefined-variable, used-before-assignment
+import os
+
+Stage0 += comment(__doc__, reformat=False)
+
+Stage0.baseimage('ubuntu:16.04')
+
+# Base dependencies
+Stage0 += python()
+Stage0 += gnu()
+Stage0 += ofed()
+
+# Additional dependencies
+ospackages = ['build-essential', 'bzip2', 'file', 'git', 'gzip',
+              'libssl-dev', 'libtool', 'lmod', 'make', 'patch',
+              'python-pip', 'python-setuptools', 'tar', 'wget',
+              'unzip', 'xz-utils']
+Stage0 += apt_get(ospackages=ospackages)
+
+# lmod setup
+Stage0 += shell(commands=[
+    'ln -s /usr/share/lmod/lmod/init/profile /etc/profile.d/lmod.sh'])
+
+# Setup and install EasyBuild
+Stage0 += shell(commands=['useradd -m easybuild',
+                          'mkdir -p /opt/easybuild',
+                          'chown easybuild:easybuild /opt/easybuild',
+                          'easy_install easybuild'])
+
+Stage0 += environment(variables={'MODULEPATH': '/opt/easybuild/modules/all:/home/easybuild/.local/easybuild/modules/all:$MODULEPATH'})
+
+easyconfig = USERARG.get('easyconfig', None)
+if easyconfig:
+     # If the easyconfig is a file in the local build context, inject it
+     # into the container image
+     if os.path.isfile(easyconfig):
+          Stage0 += copy(src=easyconfig, dest='/home/easybuild')
```

Comment:
Sorry for not jumping in earlier here...

This is definitely a good thing to do, but it may not be enough: you may need to inject some other easyconfig files as well, i.e. the ones required to build dependencies, and maybe more stuff too (like updated easyblocks, etc.)

We need a clearer picture before we can fix that though...

Reply:
Agree.  This covers the most basic case.  The most complex case you describe would currently require the user to write their own recipe (using the provided EasyBuild recipe as a starting point).  The complex case should also be considered if / when a `easybuild` building block is added.